### PR TITLE
Add permission management for repository collaborators and teams

### PR DESCRIPTION
**Pull request by banh-gao**
This PR enables the github-provider to manage the lifecycle of repository collaborators by using a new CR.

In particular it can:
- Add collaborators to a repository
- Update permissions for an existing collaborator (reconciliation)

### CODE REVIEW

**Comment by banh-gao** [comment]
Commit eb9a639a9b77deb91dc63cb4103dff103e5b5fe1 also introduces support for team permissions over a repository

File: apis/collaborator/v1alpha1/managed.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	prv1 "github.com/krateoplatformops/provider-runtime/apis/common/v1"
+)
+
+// GetCondition of this Collaborator.
+func (mg *Collaborator) GetCondition(ct prv1.ConditionType) prv1.Condition {
+	return mg.Status.GetCondition(ct)
+}
+
+// GetDeletionPolicy of this Collaborator.
+func (mg *Collaborator) GetDeletionPolicy() prv1.DeletionPolicy {
```
Comment (matteogastaldello) [inline apis/collaborator/v1alpha1/managed.go]: `provider-runtime` has been upgraded to version 0.9.0, where `deletionPolicy` has been removed. Before merging, you need to rebase the project with the main branch.

Additionally, you need to follow the folder and file structure as shown in [this commit](https://github.com/krateoplatformops/github-provider/tree/e34223ea86cefb28a991738c7f7091bf800bbb74/apis/repo/v1alpha1).

File: apis/collaborator/v1alpha1/repo_types.go
Comment (matteogastaldello) [inline apis/collaborator/v1alpha1/repo_types.go:1]: Filename should be called "collaborator_types.go"

File: apis/teamRepo/v1alpha1/repo_types.go
Comment (matteogastaldello) [inline apis/teamRepo/v1alpha1/repo_types.go:1]: Filename should be called "teamrepo_types.go"

File: Makefile
Comment (matteogastaldello) [inline Makefile:1]: The Makefile has been removed in favor of the [scripts folder](https://github.com/krateoplatformops/github-provider/tree/e34223ea86cefb28a991738c7f7091bf800bbb74/scripts).

**Review by matteogastaldello — CHANGES_REQUESTED** [review]

File: apis/collaborator/v1alpha1/managed.go (hunk 1 above)
Comment (banh-gao) [reply apis/collaborator/v1alpha1/managed.go]: Code rebased

File: apis/collaborator/v1alpha1/repo_types.go
Comment (banh-gao) [reply apis/collaborator/v1alpha1/repo_types.go:1]: file renamed

File: apis/teamRepo/v1alpha1/repo_types.go
Comment (banh-gao) [reply apis/teamRepo/v1alpha1/repo_types.go:1]: file renamed

File: Makefile
Comment (banh-gao) [reply Makefile:1]: The makefile is still present on main branch.
However I tested the changes using scripts in the folder and made changed to test new resources with them.

**Comment by banh-gao** [comment]
@matteogastaldello can you please review this PR?
Our customer is waiting for it.

File: scripts/reboot.sh
Comment (banh-gao) [inline scripts/reboot.sh:1]: this change was made because the test secret can be created only AFTER the namespace for it is created.

File: Makefile
Comment (matteogastaldello) [inline Makefile:1]: Can you please remove the Makefile? We have decided to remove it in favor of the scripts folder.

**Review by matteogastaldello — CHANGES_REQUESTED** [review]
Can you please remove the Makefile? We have decided to remove it in favor of the scripts folder.

File: Makefile
Comment (banh-gao) [reply Makefile:1]: it's in the main branch. BTW I removed it from mine.

**Review by matteogastaldello — APPROVED** [review]

**Pull request merged**